src/utils/plotSpecs/generateFrequencySpec.js

```diff
@@ -158,77 +181,39 @@ const generateSpec = (config, plotData) => {
 };
 
 const generateData = (hierarchy, properties, config) => {
-  const calculateSum = (proportionClusters, xAxisClustersIds) => {
-    let sum = 0;
-
-    if (!xAxisClustersIds.length) {
-      proportionClusters.forEach((cellSetCluster) => {
-        sum += properties[cellSetCluster.key].cellIds.size;
-      });
-      return sum;
-    }
-    proportionClusters.forEach((cellSetCluster) => {
-      const cellSetIds = Array.from(properties[cellSetCluster.key].cellIds);
-      sum += xAxisClustersIds.filter((id) => cellSetIds.includes(id)).length;
-    });
-    return sum;
+  // The key of the cell set root nodes to display on the x and y axes.
+  const cellSetGroupByRoots = {
+    x: config.xAxisGrouping,
+    y: config.proportionGrouping,
   };
 
-  const getClustersForGrouping = (name) => hierarchy.filter((cluster) => (
-    cluster.key === name))[0]?.children;
-
-  const populateData = (x, y, cluster, sum, data) => {
-    let value = y;
-    if (config.frequencyType === 'proportional') {
-      value = (y / sum) * 100;
-    }
+  // Get cell sets under the nodes.
+  const cellSets = _.mapValues(cellSetGroupByRoots, (key) => (
+    hierarchy.find((rootNode) => rootNode.key === key)?.children
+  ));
 
-    data.push({
-      x,
-      y: value,
-      c: cluster.name,
-      color: cluster.color,
-    });
+  if (!cellSets.y) {
+    return [];
+  }
 
-    return data;
-  };
+  // eslint-disable-next-line no-shadow
+  const cartesian = (...a) => a.reduce((a, b) => a.flatMap((d) => b.map((e) => [d, e].flat())));
```

Comment:
isn't like too much to add a style disable instead of just renaming the variable?  haha

Reply:
I tried and it broke, I guess I didn't rename it correctly. :D